src/PkgAuthentication.jl

```diff
@@ -19,21 +23,83 @@ abstract type Failure <: State end
 ## authentication state machine
 
 """
-    authenticate(pkgserver::AbstractString)
+    authenticate(server::AbstractString)
+
+Starts interactive (blocking) browser-based Pkg server authentication for the Pkg
+server specified by `server`. Also sets the `$(pkg_server_env_var_name)` environment
+variable to `server`.
+
+`server` must be the URL of a valid Pkg server.
 
-Starts browser based pkg-server authentication (blocking).
+## Example usage
 
-`pkgserver` must be a URL pointing to a server that provides the `/pkgserver/challenge`,
-`/pkgserver/response`, and `/pkgserver/claimtoken` endpoints.
+```julia
+julia> PkgAuthentication.authenticate("my-pkg-server.example.com")
+```
 """
 function authenticate(
     server::AbstractString;
-    force::Bool = false,
-    tries::Integer = 1,
+    auth_suffix::Union{String, Nothing} = nothing,
+    force::Union{Bool, Nothing} = nothing,
+    tries::Union{Integer, Nothing} = nothing,
+)::Union{Success, Failure}
+    ENV[pkg_server_env_var_name] = server
+    authenticate(;
+        auth_suffix = auth_suffix,
+        force = force,
+        tries = tries,
+    )
+end
+
+"""
+    authenticate()
+
+Starts interactive (blocking) browser-based Pkg server authentication for the Pkg
+server specified in the `$(pkg_server_env_var_name)` environment variable.
+
+Before calling this method, the `$(pkg_server_env_var_name)` environment variable
+must be set to the URL of a valid Pkg server.
+
+## Example usage
+
+```julia
+julia> PkgAuthentication.authenticate()
+```
+"""
+function authenticate(;
+    auth_suffix::Union{String, Nothing} = nothing,
+    force::Union{Bool, Nothing} = nothing,
+    tries::Union{Integer, Nothing} = nothing,
 )::Union{Success, Failure}
-    server = strip(server, '/')
+    if auth_suffix === nothing
+        # If the user does not provide the `auth_suffix` kwarg, we will append
+        # "/auth" at the end of the Pkg server URL.
+        #
+        # If the user does provide the `auth_suffix` kwarg, we will append
+        # "/$(auth_suffix)" at the end of the Pkg server URL.
+        auth_suffix = "auth"
+    end
+    if force === nothing
+        force = false
+    end
+    if tries === nothing
+        tries = 1
+    end
+    if tries < 1
+        throw(ArgumentError("`tries` must be greater than or equal to one"))
+    end
+
+    if !haskey(ENV, pkg_server_env_var_name)
+        msg = "The `$(pkg_server_env_var_name)` environment variable must be set"
+        throw(ErrorException(msg))
+    end
+
+    server = get_pkg_server()
+    server = rstrip(server, '/')
+    server = server * "/" * auth_suffix
```

Comment:
Nitpick, but
```suggestion
    server = string(server, "/", auth_suffix)
```

Reply:
The reason I usually prefer `*` is that it throws a `MethodError` if one of the arguments is not an `AbstractString` (which is usually the behavior that I want). In contrast, `string(...)` would silently convert non-string arguments to strings.

So, in this case, if `server` is `nothing`, we actually want to immediately throw a `MethodError`. If we use the `string(...)` version, we'd end up with `"nothing/auth"`.

Reply:
Alright, this is done now.